compiler/modules/types/core.bal

```diff
@@ -874,12 +874,9 @@ public type IntSubtypeConstraints readonly & record {|
     boolean all;
 |};
 
-// Returns `()` if `t` is not a proper, non-empty subtype of `int`.
-// i.e. returns `()` if `t` contains all or non of `int`.
-public function intSubtypeConstraints(SemType t) returns IntSubtypeConstraints? {
-    var intSubtype = intSubtype(t);
-    // JBUG can't flatten inner if-else
-    if intSubtype is boolean {
+// Returns `()` if `t` is an unconstrained int.
+public function intSubtypeConstraints(IntSubtype|true intSubtype) returns IntSubtypeConstraints? {
```

Comment:
I don't think this is an improvement.

The point of IntSubtypeConstraints is to *hide* IntSubtype for the majority of cases that don't care about it.

Also your code using this has lots of awkward casts